onnxruntime/test/providers/provider_test_utils.h

```diff
@@ -304,6 +303,7 @@ class OpTester {
     OrtValue data_;
     optional<float> relative_error_;
     optional<float> absolute_error_;
+    optional<bool> check_contents_in_order_;
   };
```

Comment:
Does this need to be wrapped with optional? What's the difference between not specified and false in behavior?

Reply:
1) I just followed the same pattern as the errors. To me this is no different than that. Not specifying a value defaults to checking contents in order (i.e.) not sorting.

2) I actually think `check_contents_in_order_` is actually more clear in its intent than `sort_output_`. `sort_output_` is our solution to checking out-of-order buffers, right ?

Reply:
Still seems a bit vague to me when I read 'check contents in order'. Does that mean I'm validating the contents (even 'contents' is a little ambiguous) are in order? 

Is 'sort output' is the implementation detail, maybe 'output is unordered' is a better name. 

---
In reply to: [331636439](https://github.com/microsoft/onnxruntime/pull/1912#discussion_r331636439) [](ancestors = 331636439)

Reply:
1. Made it plain boolean with default value (false)

2. Made the name `sort_output_`